addons/payment/models/payment_acquirer.py

```diff
@@ -227,6 +227,36 @@ def _get_default_payment_method_id(self):
         self.ensure_one()
         return self.env.ref('account.account_payment_method_manual_in').id
 
+    #=== ONCHANGE METHODS ===#
+
+    @api.onchange('state')
+    def _onchange_state(self):
+        """ Display a warning about the consequences of disabling an acquirer. """
+        self.ensure_one()
+
+        if self.state == 'disabled' and self.id:
```

Comment:
```suggestion
        if self.state == 'disabled' and self._origin.id:
```
My bad, self.id is always falsy in an onchange as NewId objects are always falsy.